docs/HowToDebugCrashWithAddr2lineTool.md

```diff
@@ -0,0 +1,163 @@
+
+In menuconfig, enable **CONFIG_DEBUG_SYMBOLS**.
+
+```
+menuconfig -> Debug Options -> Generate Debug Symbols
+```
+
+2. Add your toolchain to the system path
+It is easy to invoke addr2line if the toolchain has been included in the
+system path using the PATH environment variable.
+For example, if your ARM toolchain is located at :
+
+```
+/home/tools/gcc-arm-none-eabi-9-2019-q4-major
+```
+Then, your PATH variable must include the toolchain bin folder as follows :
+
+```
+/home/tools/gcc-arm-none-eabi-9-2019-q4-major/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+```
+
+3. Enable stackdump
+When a crash occurs in the system, it is useful to find out where exactly the crash occured and also
+the function callstack which resulted in the crash. In order to find these, we need to print the corresponding
+address values when the crash occurs. This can be done by enabling the following configs:
+
+CONFIG_ARCH_STACKDUMP
+```
+menuconfig -> Chip selection -> Dump stack on assertions
+```
+
+## How to use addr2line
+After build, use **addr2line** from linux shell.  
+Please refer to linux man page.  
+(https://man7.org/linux/man-pages/man1/addr2line.1.html)  
+Simple command is like below.  
+```
+$ addr2line -e [filepath] -a [address]
+```
+
+However, if the path to your toolchain is not configured as mentioned above, or if there are multiple toolchains installed on your system,
+then you might have to specify the correct version of the addr2line to be used. It can be done as follows:
+
+Below command specifies that ARM toolchain is to be used.
+For this to work, the ARM toolchain needs to be configured in the PATH:
+```
+$ arm-none-eabi-addr2line -e [filepath] -a [address]
+```
+
+However, if the toolchain path is not configured, or if you have mulitple versions of toolchain, then you may specify the complete path to the
+required addr2line tool as follows:
+```
+$ /home/tools/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-addr2line -e [filepath] -a [address]
+```
+
+### How to use in Binary Separation
+When binary separation is enabled, the address of text section in user binary starts from 0x0 for relocation.  
+After relocation, its real address is allocated and those address can be used for debugging.  
+For this, some steps are required like below.  
+
+1. Enable these configs:
+
+	CONFIG_SAVE_BIN_SECTION_ADDR  
+	```
+	menuconfig -> Binary Loader -> Save binary section address
+	```
+
+	CONFIG_DEBUG_BINFMT_INFO
+	```
+	menuconfig -> Debug Options -> Enable Debug Features -> Binary Loader Debug Features -> Binary Loader Informational Output
+	```
+
+2. Check the user binary's text address from the printed log.  
+
+3. Calculate the address based on No.2.  
+For example, let assume that user binary text address is 0x1000000, and symbol address ix 0x1200000.  
+Then subtract 0x1000000 from 0x1200000 and pass this address for -a option's parameter.
+	```
+	$ addr2line -e [filepath] -a 0x200000
+	```
+
```

Comment:
Could you add our assert message for example and explain what contents are there?

Reply:
I have added the example at the end. Please let me knew if anything else is required.